examples/start_stop/main.cc

```diff
@@ -0,0 +1,176 @@
+);
+
+DEFINE_string(
+  stream_addr, "localhost",
+  "Where the device should stream the image data. "
+  "**Note** This flag tells the device where to send the image data, so this "
+  "address should be from the device's viewpoint."
+);
+
+DEFINE_string(
+  run_duration, "120",
+  "How long (in seconds) to run this program before disconnect. "
+);
+
+DEFINE_string(
+  log_path, "log.txt",
+  "file path of the output log file."
+);
+
+using namespace zen::api;
+
+
+namespace {
+
+bool
+ModeExists(ZenApi::Mode mode, std::vector<ZenApi::Mode> modes)
+{
+  if (std::find(modes.begin(), modes.end(), mode) != modes.end()){
+    return true;
+  }
+  else {
+    return false;
+  }
+}
+
+void
+SpinImages(int* image_counter)
+{
+  while (auto next_image = ZenApi::NextImage()) {
+    (void)next_image;
+    *image_counter += 1;
+  }
+}
+
+void
+SpinPoints(int* pc_counter)
+{
+  while (auto next_points = ZenApi::NextTrackerState()) {
+    (void)next_points;
+    *pc_counter += 1;
+  }
+}
+
+void
+SpinLogs(std::ofstream* file)
+{
+  while (auto next_log = ZenApi::NextLogMessage()) {
+    std::cout << next_log->canonical_message() << std::endl;
+    if (next_log->severity() > 0){
+      (*file) << next_log->canonical_message() << std::endl;
+    }
+  }
+}
+
+void
+SpinHK()
+{
+  while (auto next_hk = ZenApi::NextHousekeepingReport()) {
+    (void)next_hk;
+  }
+}
+
+} ///< \namespace
+
+
+int
+main(int argc, char* argv[])
+{
+  ZenApi::Init(&argc, &argv);
+
+  int pc_counter = 0;
+  int image_counter = 0;
+
+  std::ofstream log_file;
+  log_file.open(FLAGS_log_path);
+
+  auto default_telem_ports = ZenApi::TelemPortOptions();
+  ZenApi::Connect(FLAGS_device_addr, default_telem_ports);
+
+  auto default_data_ports = ZenApi::DataPortOptions();
+  ZenApi::Bind(default_data_ports);
+
+  // verify the input mode actually exists on ZPU
+  auto modes = ZenApi::ListModes();
+  bool mode_exists = ModeExists(FLAGS_imaging_mode, modes);
+  if (!mode_exists){
+    std::cout << FLAGS_imaging_mode <<
+                " does not exist on ZPU. Available modes are: " << std::endl;
+
+    for (const auto& mode : modes) {
+      std::cout << mode << std::endl;
+    }
+
+    return EXIT_FAILURE;
+  }
+
+  // start the radars
+  auto default_install_options = ZenApi::InstallOptions();
+  default_install_options.install_path = FLAGS_install_path;
+  ZenApi::Start(FLAGS_imaging_mode, default_install_options, FLAGS_stream_addr);
+
+  std::cout << "radar started" << std::endl;
+
+  // subscribe to radar data
+  ZenApi::SubscribeImages();
+  ZenApi::SubscribeTrackerStates();
+  ZenApi::SubscribeLogMessages();
+  ZenApi::SubscribeHousekeepingReports();
+
+  // listen to data
+  auto image_reader = std::thread(SpinImages, &image_counter);
+  auto points_reader = std::thread(SpinPoints, &pc_counter);
+  auto log_reader = std::thread(SpinLogs, &log_file);
+  auto hk_reader = std::thread(SpinHK);
+
+  std::this_thread::sleep_for(
+      std::chrono::seconds(std::stoi(FLAGS_run_duration)));
```

Comment:
Suggest to change the type from string to int32, and then won't need `std::stoi(...)`. See previous comment.